### PR TITLE
Move HeatPump:PlantLoop:EIR:* sizing calculations

### LINKED ISSUE #9733: Autosized source-side air flow rate for air-source HeatPump:PlantLoop:EIR:* object is negative
Issue overview
--------------
The autosized source-side air flow rate of air-source `HeatPump:PlantLoop:EIR:*` objects is negative. Here's an excerpt of simulation outputs that shows that the air flow rate is negative (about -15 kg/s):

image

### Details
Some additional details for this issue (if relevant):
 - Platform: Windows 10
 - Version of EnergyPlus: tested 22.1 and 22.2

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Defect file added: [defect_model.zip](https://github.com/NREL/EnergyPlus/files/9976176/defect_model.zip)
 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by lymereJ**
Pull request overview
---------------------
 - Fixes #9733 

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [x] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [x] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [x] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [x] If changes fix a defect, the fix should be demonstrated in plots and descriptions

### Reviewer
This will not be exhaustively relevant to every PR.
 - [x] Perform a Code Review on GitHub
 - [x] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [x] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [x] CI status: all green or justified
 - [x] Check that performance is not impacted (CI Linux results include performance check)
 - [x] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

**Comment by lymereJ** [comment]
The figure below shows the results for the defect file (converted to 22.2) when the proposed changes are applied.
image

File: src/EnergyPlus/PlantLoopHeatPumpEIR.cc [hunk 1]
```diff
@@ -392,12 +401,6 @@ void EIRPlantLoopHeatPump::getDesignCapacities(
     EnergyPlusData &state, const PlantLocation &calledFromLocation, Real64 &MaxLoad, Real64 &MinLoad, Real64 &OptLoad)
 {
     if (calledFromLocation.loopNum == this->loadSidePlantLoc.loopNum) {
-        this->sizeLoadSide(state);
-        if (this->waterSource) {
-            this->sizeSrcSideWSHP(state);
-        } else if (this->airSource) {
-            this->sizeSrcSideASHP(state);
-        }
```
Comment (lymereJ) [inline src/EnergyPlus/PlantLoopHeatPumpEIR.cc:408]: The `referenceCapacity` needed by `sizeSrcSideASHP` to size `sourceSideDesignVolFlowRate` is only set when `state.dataPlnt->PlantFirstSizesOkayToFinalize` is `true` which was never the case when `EIRPlantLoopHeatPump::getDesignCapacities` was called which resulted in the `referenceCapacity` used for sizing calculations to always be -99999 which always resulted in a negative autosized source side air flow rate. The proposed changes move this block in `EIRPlantLoopHeatPump::onInitLoopEquip` which gets called through the different passes of `InitializeLoops` when `state.dataPlnt->PlantFirstSizesOkayToFinalize` is set to true.

`ChillerElectricEIR` follows a similar approach.

File: tst/EnergyPlus/unit/PlantLoopHeatPumpEIR.unit.cc [hunk 2]
```diff
@@ -2565,6 +2565,111 @@ TEST_F(EnergyPlusFixture, TestSizing_HardsizedFlowAutosizedCoolingWithCompanion_
     EXPECT_NEAR(expectedCapacity, tmpOpt, 0.001);
 }
 
+TEST_F(EnergyPlusFixture, TestSizing_AutosizedFlowWithCompanion_AirSource)
```
Comment (mjwitte) [inline tst/EnergyPlus/unit/PlantLoopHeatPumpEIR.unit.cc:2637]: I merged in the latest develop and now this unit test is failing (on windows).
```
error: The difference between expectedClgSourceFlow and thisCoolingPLHP->sourceSideDesignVolFlowRate is 100085.71000000001, which exceeds 0.1, where
expectedClgSourceFlow evaluates to 86.709999999999994,
thisCoolingPLHP->sourceSideDesignVolFlowRate evaluates to -99999, and
0.1 evaluates to 0.10000000000000001.
```

Comment (mjwitte) [reply tst/EnergyPlus/unit/PlantLoopHeatPumpEIR.unit.cc:2637]: Hmm, I see why. I started over with this branch, then merged in develop which reverts the change from this branch for some reason. If you can get that fixed, then I'll retest.

**Comment by mjwitte** [comment]
Also, I don't see the autosized source side volume flow rate getting reported to the eio component sizing output. Seems like it should. @rraustad

**Comment by rraustad** [comment]
All autosized fields should be reported to eio as either User Specified (hard sized) or Design Size (autosized) or both if different by 10% (for comparison). New sizing routines will do this automatically.

**Comment by Myoldmopar** [comment]
@mjwitte I'll leave this to you for now since you worked on it yesterday, just let me know if you want me to do any additional testing.  Moving the code to follow the chiller pattern makes sense to me.

**Comment by lymereJ** [comment]
@mjwitte - The last CI results now show diffs. They're due to the fact that I added reporting of the source side (volume) air flow rate which was previously missing.

**Review by mjwitte — APPROVED** [review]
All good here. Thanks @lymereJ

**Comment by lincolnharmer** [comment]
Thanks all. I just tested this and things are now working as expected.
image

**Pull request merged**